airflow/plugins_manager.py

```diff
@@ -113,7 +113,7 @@ class EntryPointSource(AirflowPluginSource):
     """Class used to define Plugins loaded from entrypoint."""
 
     def __init__(self, entrypoint: importlib_metadata.EntryPoint, dist: importlib_metadata.Distribution):
-        self.dist = dist.metadata['name']
+        self.dist = dist.metadata['Name']
```

Comment:
Python pacakge metadata keys are case-insensitive, I use `Name` ([the case used by the standard](https://packaging.python.org/en/latest/specifications/core-metadata/)) everywhere so tests are easier to write.